lib/fog/libvirt/requests/compute/list_domains.rb

```diff
@@ -48,6 +48,23 @@ def boot_order xml
           xml_elements(xml, "domain/os/boot", "dev")
         end
 
+        # Foreman expects the firmware to be 'uefi_sb' if SB is enabled
+        def firmware(xml)
+          firmware_type = xml_elements(xml, "domain/os", "firmware").first || 'bios'
+          return 'uefi_sb' if firmware_type == 'efi' && secure_boot_enabled?(xml)
+
+          firmware_type
+        end
+
+        def secure_boot_enabled?(xml)
+          enabled_features = xml_elements(xml, "domain/os/firmware/feature")
+                               .select { |feature| feature[:enabled] == 'yes' }
+                               .map { |feature| feature[:name] }
+
+          required_features = ['secure-boot', 'enrolled-keys']
```

Comment:
AFAIK you don't need `enrolled-keys` for secure-boot. That's just the default set of keys, but users can define their own keys and sign their own kernels.

Reply:
https://libvirt.org/kbase/secureboot.html#additional-information says that without `enrolled-keys` it allows running unsigned code, but I wonder if that's really true. This last bit clarifies with how I think it's supposed to work:

> The main difference between using a build of EDKII that has Secure Boot support but without keys enrolled and one that doesn't have Secure Boot support at all is that, with the former, you could enroll your own keys and securely run an operating system that you've built and signed yourself. If you are only planning to run existing, off-the-shelf operating system images, then the two configurations are functionally equivalent.

So the idea is that `enrolled-keys` contains the default keys, but if you don't have or want those you can enroll your own keys.